File: algorithm/src/main/java/baekjoon/p10419/Main.java
package baekjoon.p10419;

import java.util.Scanner;

public class Main {

    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int t = sc.nextInt();

        while (t-- > 0) {
            System.out.println(solve(sc.nextInt()));
        }
    }

    private static int solve(int d) {
        int t = 0;

        while (t * (t + 1) <= d) {
            t++;
        }

        return (t - 1);
    }

}

File: algorithm/src/main/java/baekjoon/p10419/Main.kt
package baekjoon.p10419

fun main(args: Array<String>) {
    repeat(readLine()!!.toInt()) {
        println(solve(readLine()!!.toInt()))
    }
}

private fun solve(d: Int): Int {
    var t = 0

    while (t * (t + 1) <= d) {
        t++
    }

    return (t - 1)
}